battlebots/database/access.py

```diff
@@ -37,7 +37,8 @@ def remove_bot(user, botname):
                         % (user.nickname, botname, code_dir))
         pass
 
-    bot = session.query(Bot).filter_by(user=user, name=botname).one_or_none()
```

Comment:
Correct, but I'm thinking that maybe that isn't a problem since you close the session at the end of each web request. It has the benefit of not opening+closing a session for each query.

On the other hand, this might cause problem with the ranker. Or not. I think we have to decide once and for all how we handle sessions: (1) at the low-level query/add/delete, or (2) at the beginning and end of a web request, and at the beginning and end of a battle.